eventdiscovery-sdk/src/main/java/com/schedjoules/eventdiscovery/framework/eventlist/EventListFragment.java

```diff
@@ -300,28 +299,17 @@ public void onDestroy()
 
     private GeoLocation location()
     {
-        if (getArguments() != null && getArguments().containsKey(EXTRA_GEOLOCATION))
+        if (new Argument<>(Keys.GEO_LOCATION, getArguments()).isPresent())
         {
             // TODO Save location and update Toolbar title with name when Event Discovery for input geo-location is actually supported
-            return getArguments().getParcelable(EXTRA_GEOLOCATION);
-        }
-        else
-        {
-            return new SharedPrefLastSelectedPlace(getContext()).get().geoLocation();
+            throw new UnsupportedOperationException("Discovery for a geo location not supported yet.");
         }
+        return new SharedPrefLastSelectedPlace(getContext()).get().geoLocation();
     }
 
 
     private DateTime startAfter()
     {
-        if (getArguments() != null && getArguments().containsKey(EXTRA_START_AFTER_TIMESTAMP))
-        {
-            long startAfterTimeStamp = getArguments().getLong(EXTRA_START_AFTER_TIMESTAMP, 0);
-            return new DateTime(startAfterTimeStamp);
-        }
-        else
-        {
-            return DateTime.nowAndHere();
-        }
+        return new Argument<>(Keys.DATE_TIME_START_AFTER, getArguments()).value(DateTime.nowAndHere());
```

Comment:
As mentioned above, I think we are talking about a factory here indeed, to create an object lazily based on conditions. Optional would be a misuse for this, I think.